### ISSUE TITLE
Plugin crashes on Ubuntu 15.10, Kodi 15.2

### ISSUE THREAD

**Issue by Josef-Friedrich** [cccc/plugin.video.media-ccc-de#6]
# Versions:

* Ubuntu 15:10
* Kodi 15.2
* Plugin: 0.0.4

# Error showing in the GUI:

```
CCC-TV (media.ccc.tk) error 
Check log for more informations
```

# Error in the log file

```
13:28:02 T:140576881395456   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.RuntimeError'>
                                            Error Contents: osrandom engine already registered
                                            Traceback (most recent call last):
                                              File "/home/jf/.kodi/addons/plugin.video.media-ccc-de/addon.py", line 10, in <module>
                                                import resources.lib.http as http
                                              File "/home/jf/.kodi/addons/plugin.video.media-ccc-de/resources/lib/http.py", line 1, in <module>
                                                import requests
                                              File "/home/jf/.kodi/addons/script.module.requests/lib/requests/__init__.py", line 53, in <module>
                                                from .packages.urllib3.contrib import pyopenssl
                                              File "/home/jf/.kodi/addons/script.module.requests/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 54, in <module>
                                                import OpenSSL.SSL
                                              File "/usr/lib/python2.7/dist-packages/OpenSSL/__init__.py", line 8, in <module>
                                                from OpenSSL import rand, crypto, SSL
                                              File "/usr/lib/python2.7/dist-packages/OpenSSL/rand.py", line 11, in <module>
                                                from OpenSSL._util import (
                                              File "/usr/lib/python2.7/dist-packages/OpenSSL/_util.py", line 7, in <module>
                                                binding = Binding()
                                              File "/usr/lib/python2.7/dist-packages/cryptography/hazmat/bindings/openssl/binding.py", line 63, in __init__
                                                self._ensure_ffi_initialized()
                                              File "/usr/lib/python2.7/dist-packages/cryptography/hazmat/bindings/openssl/binding.py", line 95, in _ensure_ffi_initialized
                                                cls._register_osrandom_engine()
                                              File "/usr/lib/python2.7/dist-packages/cryptography/hazmat/bindings/openssl/binding.py", line 70, in _register_osrandom_engine
                                                raise RuntimeError("osrandom engine already registered")
                                            RuntimeError: osrandom engine already registered
                                            -->End of Python script error report<--
13:28:02 T:140577722099776   ERROR: GetDirectory - Error getting plugin://plugin.video.media-ccc-de/
13:28:02 T:140577722099776   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.media-ccc-de/) failed
13:28:02 T:140576881395456  NOTICE: Thread BackgroundLoader start, auto delete: false
```

**Comment by Josef-Friedrich** [comment]
I had to upgrade the package _python-cryptography_ to make the pluging work again:

```
apt-get install build-essential libssl-dev libffi-dev python-dev
pip install cryptography --upgrade
```

**Comment by TobiX** [comment]
Unfortunately, this is a bug in python-cryptography (pyca/cryptography#2287 to be precise, fixed in version 1.1(?)) - I don't think we can do anything from this add-on to fix the issue... If anyone thinks I can do anything to fix this inside the add-on, please reopen this issue.
